ui/src/pages/Results.js

```diff
@@ -1,20 +1,42 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { usePetStats } from "../resources/pets";
 import Main from "../layouts/Main";
 import Spinner from "../components/Spinner";
 import PetStats from "../components/PetStats";
 
+// TODO: allow for more buckets
+const timeBuckets = ["1m", "1h", "1d"];
+
 function Results() {
-  const { data: byType, isLoading: isLoadingType } = usePetStats("type");
-  const { data: byClient, isLoading: isLoadingClient } = usePetStats("client");
+  const [timeBucket, setTimeBucket] = useState(timeBuckets[0]);
+
+  const { data: byType, isLoading: isLoadingType } = usePetStats({
+    groupBy: "type",
+  });
+
+  const { data: byClient, isLoading: isLoadingClient } = usePetStats({
+    groupBy: "client",
+  });
+
+  const { data: byTime, isLoading: isLoadingTime } = usePetStats({
+    groupBy: "time",
+    timeBucket,
+  });
 
   return (
     <Main>
-      {isLoadingType || isLoadingClient ? (
+      {isLoadingType || isLoadingClient || isLoadingTime ? (
```

Comment:
Which one looks better? Wait for the whole lot or load the page and have spinners for each chart? This is getting not very self-contained...